src/main/java/com/github/manolo8/darkbot/core/manager/HangarManager.java

```diff
@@ -0,0 +1,222 @@
+                this.lastChangeHangar = System.currentTimeMillis();
+            }
+
+        } else {
+            return false;
+        }
+
+        return true;
+    }
+
+    private String getDataInventory(String params){
+        String data = null;
+        HttpURLConnection conn = null;
+        InputStream inputStream;
+        String instance = this.main.statsManager.instance, sid = this.main.statsManager.sid;
+        if (instance == null || instance.isEmpty() || sid == null || sid.isEmpty()) return data;
+
+        String url = instance + params;
+        try {
+            conn = (HttpURLConnection) new URL(url)
+                    .openConnection();
+            conn.setInstanceFollowRedirects(false);
+            conn.setRequestProperty("Content-Type","application/x-www-form-urlencoded");
+            conn.setRequestProperty("Cookie", "dosid=" + sid);
+            int responseCode = conn.getResponseCode();
+            if (200 <= responseCode && responseCode <= 299) {
+                inputStream = conn.getInputStream();
+            } else {
+                inputStream = conn.getErrorStream();
+            }
+
+            BufferedReader in = new BufferedReader(new InputStreamReader(inputStream));
+            StringBuilder responseb = new StringBuilder();
+            String currentLine;
+
+            while ((currentLine = in.readLine()) != null)
+                responseb.append(currentLine);
+
+            in.close();
+            inputStream.close();
+
+            byte[] base64Decode = Base64.getDecoder().decode(responseb.toString());
+            data = new String(base64Decode, "UTF-8");
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            if (conn != null ){
+                conn.disconnect();
+            }
+        }
+
+        return data;
+    }
+
+    public void updateDrones() {
+        try {
+            String hangarID = getActiveHangar();
+
+            if (hangarID != null) {
+                String decodeParams = "{\"params\":{\"hi\":" + hangarID + "}}";
+                String encodeParams = Base64.getEncoder().encodeToString(decodeParams.getBytes("UTF-8"));
+                String url = "/flashAPI/inventory.php?action=getHangar&params="+encodeParams;
+                String json = getDataInventory(url);
+                JsonObject object =  new JsonParser().parse(json).getAsJsonObject();
+                JsonObject data = object.get("data").getAsJsonObject();
+                JsonObject ret  = data.get("ret").getAsJsonObject();
+                JsonArray hangardata = ret.get("hangars").getAsJsonArray();
+
+                for (JsonElement hangar : hangardata.getAsJsonArray()) {
+                    boolean active = hangar.getAsJsonObject().get("hangar_is_active").getAsBoolean();
+
+                    if (active) {
+                        JsonArray dronesArray = hangar.getAsJsonObject().get("general").getAsJsonObject().get("drones").getAsJsonArray();
```

Comment:
Again, shouldn't you just use POJOs instead of Json datatypes?

Reply:
In this case it is to have more friendly variables, but if I can map it to an object directly if you prefer.